verification/src/http_server/handlers/verification/solidity/handlers.rs

```diff
@@ -0,0 +1,106 @@
+    #[error("{0:#}")]
+    Compilation(#[from] CompilersError),
+    #[error("No contract could be verified with provided data")]
+    NoMatchingContracts,
+}
+
+pub(crate) async fn compile_and_verify_handler<T: Fetcher>(
+    compilers: &Compilers<T>,
+    mut input: CompileAndVerifyInput<'_>,
+    bruteforce_bytecode_hashes: bool,
+) -> Result<VerificationResponse, actix_web::Error>
+where
+    <T as Fetcher>::Error: Debug + Display,
+{
+    let bruteforce_metadata = if !bruteforce_bytecode_hashes {
+        Vec::from([input.compiler_input.settings.metadata.clone()])
+    } else if pre_v0_6_0(&input.compiler_version) {
+        Vec::from([None])
+    } else {
+        Vec::from(BYTECODE_HASHES.map(|hash| Some(SettingsMetadata::from(hash))))
+    };
+
+    for metadata in bruteforce_metadata {
+        input.compiler_input.settings.metadata = metadata;
+        match compile_and_verify(compilers, &input).await {
+            Ok(verification_success) => {
+                let verification_result = VerificationResult::from((
+                    input.compiler_input,
+                    input.compiler_version,
+                    verification_success,
+                ));
+                return Ok(VerificationResponse::ok(verification_result));
+            }
+            Err(CompileAndVerifyError::VerifierInitialization(err)) => {
+                return Err(error::ErrorBadRequest(err))
+            }
+            Err(CompileAndVerifyError::Compilation(CompilersError::Fetch(err))) => {
+                return Err(error::ErrorInternalServerError(err))
+            }
+            Err(CompileAndVerifyError::Compilation(CompilersError::Internal(err))) => {
+                return Err(error::ErrorInternalServerError(err))
+            }
+            err @ Err(CompileAndVerifyError::Compilation(CompilersError::Compilation(_))) => {
+                return Ok(VerificationResponse::err(err.expect_err("Matched to Err")))
+            }
+            // Try other bytecode hashes if there is no matching contracts
+            Err(CompileAndVerifyError::NoMatchingContracts) => {}
+        }
+    }
+    // In case of any other error the execution will not get to this point
+    Ok(VerificationResponse::err(
+        CompileAndVerifyError::NoMatchingContracts,
+    ))
+}
+
+async fn compile_and_verify<T: Fetcher>(
+    compilers: &Compilers<T>,
+    input: &CompileAndVerifyInput<'_>,
+) -> Result<VerificationSuccess, CompileAndVerifyError>
+where
+    <T as Fetcher>::Error: Debug + Display,
+{
+    let verifier = Verifier::new(input.creation_tx_input, input.deployed_bytecode)?;
+    let compiler_output = compilers
+        .compile(&input.compiler_version, &input.compiler_input)
+        .await?;
+    verifier
+        .verify(compiler_output)
+        .ok_or(CompileAndVerifyError::NoMatchingContracts)
+}
+
+fn pre_v0_6_0(compiler_version: &CompilerVersion) -> bool {
+    let last_non_0_6_0_version =
```

Comment:
we can just compare semvers, and also semvers implement constrains, so you can do something like
```rust
let req = VersionReq::parse("<0.6.0").unwrap();
if req.matches(compiler_version.version()) {
```
also I don't like the name of the function and suggest just remove the function and do the versionreq variant